### PR TITLE
feat(ui5-button): add accessible-role api

### PR DESCRIPTION
**Pull request by tsanislavgatev**
fixes: https://github.com/SAP/ui5-webcomponents/issues/7612

### CODE REVIEW

File: packages/main/src/Button.ts [hunk 1]
```diff
@@ -246,6 +247,18 @@ class Button extends UI5Element implements IFormElement, IButton {
...
+	 * Describes the accessibility role of the button.
+	 *
+	 * NOTE: Use link role only with a press handler, which performs a navigation. In all other scenarios the default button semantics is recommended.
+	 *
+	 * @default "Button"
+	 * @public
+	 * @since 1.23
+	 */
+	@property({ type: ButtonAccessibleRole, defaultValue: ButtonAccessibleRole.Button })
+	accessibleRole!: string;
```
Comment (hinzzx) [inline packages/main/src/Button.ts:260]: The type here should be `${ButtonAccessibleRole}` by convention. Same way as the above property `type`.

**Review by hinzzx — CHANGES_REQUESTED** [review]

File: packages/main/src/Button.ts [hunk 2]
```diff
@@ -246,6 +247,18 @@ class Button extends UI5Element implements IFormElement, IButton {
 	@property({ type: ButtonType, defaultValue: ButtonType.Button })
 	type!: `${ButtonType}`;
 
+	/**
+	 * Describes the accessibility role of the button.
+	 *
+	 * NOTE: Use link role only with a press handler, which performs a navigation. In all other scenarios the default button semantics is recommended.
```
Comment (hinzzx) [inline packages/main/src/Button.ts:253]: Very minor one:
> ...button semantics is recommended

semantics **are** recommended.

**Review by hinzzx — APPROVED** [review]
Looks good to me.

**Review by nnaydenow — CHANGES_REQUESTED** [review]
As discussed offline

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**